src/condition-hierarchy.lisp

```diff
@@ -8,13 +8,13 @@
 ;;; documentation strings due to the repetitiveness of the contents of this
 ;;; file.
 
-(define-condition warning (condition) ())
+(define-condition-internal warning (condition) ())
```

Comment:
None of these changes from `define-condition` to `define-condition-internal` should be required. Since `condition` already names a condition type, we should be able to use `define-condition` instead of `define-condition-internal`.

Why were they added?

Reply:
When running the test system with DEFINE-CONDITION here, I get INVALID INITIALIZATION ARGUMENTS errors like:
```common-lisp
;; Invalid initialization arguments:
;;   :EXPECTED-TYPE, :DATUM
;; in call for class #<STANDARD-CLASS PORTABLE-CONDITION-SYSTEM::CASE-FAILURE>.
;;    [Condition of type SB-PCL::INITARG-ERROR]
;; See also:
;;   Common Lisp Hyperspec, 7.1.2 [:section]
```

The way COERCE-CONDITION-SUPERTYPES was defined, it was filtering out PCS:SERIOUS-CONDITION and PCS:ERROR from the PCS::CASE-FAILURE condition, for example. But they were superclasses of PCS::CASE-FAILURE before the changes. I assume SUBTYPEP in COERCE-CONDITION-SUPERTYES couldn't tell the relations at this point?

Reply:
Yes, this is what I feared - that `subtypep`, when run at compile-time, does not have information about class definitions. If left as-is, this is going to propagate into users' code, and this may break in weird ways for them where loading a file is going to have different semantics than compiling it.

A possible (and better!) solution I can imagine is adding a check into the class (re)definition process itself - e.g. a `:before` method on `ensure-class-using-class` can ensure that all superclasses are `condition` or proper subclasses of `condition`.

Reply:
Yes, this isn't acceptable. Thank you for the feedback.

Reply:
I'm following your suggestion about `ensure-class-using-class`, and came across a doubt. What should be the behavior of:

```common-lisp
(define-condition foo-condition () ())

(defclass bar (foo-condition) ())
```
?

sbcl, for example, signals an INVALID-SUPERCLASS

Reply:
I guess we can either leave this behavior undefined (because people are supposed to unconditionally use `define-condition` to define condition types), or signal our own error in such a situation.